api/src/main/java/bio/terra/tanagra/service/search/Filter.java

```diff
@@ -74,5 +76,67 @@ public static ArrayFunction create(List<Filter> operands, Operator operator) {
     }
   }
 
-  // TODO implement a relationship filter tht allows chaining to another entity.
+  /**
+   * A {@link Filter} that introduces a constraint that there must be a relationship to another
+   * entity.
+   *
+   * <p>This introduces a new {@link Variable} and implicitly binds it to an entity in {@link
+   * #boundAttribute()}. The bound and related entities should have a relationship across the bound
+   * attributes. The outer variable may be any variable introduced higher in the outer filter scope.
+   *
+   * <p>This is equivalent to the pseudo-SQL ":outerAttribute IN (SELECT :boundAttribute FROM
+   * :boundAttribute.entity AS :boundAttribute.variable WHERE :filter)".
+   *
+   * <p>Or in words ∃x∈boundEntity( x.boundAttribute = outerAttribute ∧ filter): there exists an
+   * instance 'x' of {@link #boundAttribute()}'s Entity where x's {@link #boundAttribute()} equals
+   * {@link #outerAttribute()} and {@link #filter}.
```

Comment:
Thanks for the updates. One thing I'm still a little bit confused by:

Here it says "boundAttribute equals outerAttribute and filter". I'm having difficulty following what it means for `boundAttribute` to "equal" a filter.

Reply:
parsing ambiguity, added some parens.